UC-Berkeley-Extension/global
# Workbench access

## Asana Ticket(s)
- See ticket [Permissions to Publish](https://app.asana.com/0/1174970560648905/1198160276496235/f).

## Description
- Sets up Workbench Access for control of editorial content.


## Testing instructions
1. Checkout this branch
1. From project root, run `composer install`
1. Build the site
  * From inside the VM, run `/var/www/html/scripts/setup.sh`
1. Login and go to https://global.ddev.site/admin/structure/taxonomy/manage/editorial_section/overview
  * Note that 5 taxonomy terms exist
1. Go to https://global.ddev.site/node/2
  * Note the `Editorial Section` is set to `Site Administrators`
  * Note the `Workbench Access information` block shows the same
1. Go to https://global.ddev.site/admin/structure/types/manage/article/fields
  * Note the `Editorial Section` field -- field_term_ref_editorial_section
  * Note that this field has been added to all content types and:
    - Is a single-value select list
    - Is required 
    - Defaults to 'Site Administrators`
1. Go to https://global.ddev.site/admin/people/permissions
  * Note the default permissions for `authenticated users` to:
    - create, edit, and delete their own content and media
    - view revisions
    - Use Workbench
    - Use Workbench Access
1. Note the `Workbench` toolbar element
1. Go to https://global.ddev.site/admin/workbench
   * See the content lists show Editorial Section
   * Note that both are set to `Site Administrators`
1. As admin, create content and confirm you can change its section to any of the 5 terms.
1. Go to https://global.ddev.site/admin/config/workflow/workbench_access
  * Note the `Editorial Section` access scheme.
  * Click the `Edit` operation
  * Note that it is set to use the `Editorial Sections` taxonomy and field for all content types.
1. Click the `Sections` tab
  * Note that no assignments have been made.

1. Create a new user
1. Assign that user (not role) to the `Career Services` section at https://global.ddev.site/admin/config/workflow/workbench_access/editorial_section/sections/2/users
1. Logout and login as that user
1. Observe the limited toolbar
1. Observe the Workbench toolbar element
1. Create a basic page.
1. Observe that you can only assign it to the `Career Services` section.

# Notes

* The admin user does not need to be assigned to sections, since they are automatically in all sections.
